### PR TITLE
Working group weight annotations

### PR DESCRIPTION
**Pull request by conectado**
This PR includes:

* Fix `working_group` benchmark in Olympia
* Add `working_group` pallet to benchmark
  * **Note:** I used `ForumWorkingGroup` but for what I've seen which instance we should use to benchmark is arbitrary
* Add weight annotations for `working_group` pallet

### CODE REVIEW

**Comment by iorveth** [comment]
Just one comment here. 
Wouldn`t it be safer to have separate benchmarks for all working group instances on node runtime level?
My primarily concern is that we can probably have individual constraint limits for working group instances and that can result in different worst-case scenarios. https://github.com/Joystream/joystream/blob/olympia/runtime/src/lib.rs#L620

**Comment by conectado** [comment]
> Just one comment here.
> Wouldn`t it be safer to have separate benchmarks for all working group instances on node runtime level?
> My primarily concern is that we can probably have individual constraint limits for working group instances and that can result in different worst-case scenarios. https://github.com/Joystream/joystream/blob/olympia/runtime/src/lib.rs#L620

I agree. I don't know if it's possible however to run multiple benchmarks for the same pallet. I will try.

**Comment by conectado** [comment]
> > Just one comment here.
> > Wouldn`t it be safer to have separate benchmarks for all working group instances on node runtime level?
> > My primarily concern is that we can probably have individual constraint limits for working group instances and that can result in different worst-case scenarios. https://github.com/Joystream/joystream/blob/olympia/runtime/src/lib.rs#L620
> 
> I agree. I don't know if it's possible however to run multiple benchmarks for the same pallet. I will try.

Can be done, will upload the changes ASAP.

**Comment by bedeho** [comment]
Nice observation @iorveth .

**Comment by conectado** [comment]
> > > Just one comment here.
> > > Wouldn`t it be safer to have separate benchmarks for all working group instances on node runtime level?
> > > My primarily concern is that we can probably have individual constraint limits for working group instances and that can result in different worst-case scenarios. https://github.com/Joystream/joystream/blob/olympia/runtime/src/lib.rs#L620
> > 
> > 
> > I agree. I don't know if it's possible however to run multiple benchmarks for the same pallet. I will try.
> 
> Can be done, will upload the changes ASAP.

After implementing the way I found to do it, I found a problem where the weight-generated files can't be used without manually modifying them I will research more.

**Comment by conectado** [comment]
> > > > Just one comment here.
> > > > Wouldn`t it be safer to have separate benchmarks for all working group instances on node runtime level?
> > > > My primarily concern is that we can probably have individual constraint limits for working group instances and that can result in different worst-case scenarios. https://github.com/Joystream/joystream/blob/olympia/runtime/src/lib.rs#L620
> > > 
> > > 
> > > I agree. I don't know if it's possible however to run multiple benchmarks for the same pallet. I will try.
> > 
> > 
> > Can be done, will upload the changes ASAP.
> 
> After implementing the way I found to do it, I found a problem where the weight-generated files can't be used without manually modifying them I will research more.

Until [this PR](https://github.com/paritytech/substrate/pull/7669) which solves this problem lands in substrate and we migrate to a version that includes it, we will either use the worst-case instance for benchmarking and if there's no such obvious instance we will include a mock instance representing the worst-case.

For this PR the worst-case instance is `ContentDirectoryWorkingGroup`

File: runtime-modules/working-group/src/lib.rs [hunk 1]
```diff
@@ -469,7 +563,23 @@ decl_module! {
...
-        #[weight = 10_000_000] // TODO: adjust weight
+        /// # <weight>
+        ///
+        /// ## Weight
+        /// `O (P)` where:
+        /// - `P` is the length of `penalty.slashing_text`
+        /// - DB:
+        ///    - O(1) doesn't depend on the state or parameters
+        /// # </weight>
+        #[weight = WeightInfoWorkingGroup::<T, I>::terminate_role_lead(
```
Comment (shamil-gadelshin) [inline runtime-modules/working-group/src/lib.rs]: Consider moving the weight-code to a separate function.

File: runtime-modules/working-group/src/lib.rs [hunk 2]
```diff
@@ -767,7 +941,23 @@ decl_module! {
...
+        ///
+        /// ## Weight
+        /// `O (S)` where:
+        /// - `S` is the length of the contents of `status_text` when it is not none
+        ///
+        /// - DB:
+        ///    - O(1) doesn't depend on the state or parameters
+        /// # </weight>
+        #[weight = WeightInfoWorkingGroup::<T, I>::set_status_text(
+            if let Some(status_text) = status_text {
```
Comment (shamil-gadelshin) [inline runtime-modules/working-group/src/lib.rs]: Consider changing this snippet to a functional style: using Option.map with unwrap_or_default().

File: runtime-modules/working-group/src/lib.rs [hunk 3]
```diff
@@ -58,6 +59,36 @@ pub use checks::{ensure_origin_is_active_leader, ensure_worker_exists, ensure_wo
 use common::origin::ActorOriginValidator;
 use staking_handler::StakingHandler;
 
+type WeightInfoWorkingGroup<T, I> = <T as Trait<I>>::WeightInfo;
+
+/// working_group WeightInfo
```
Comment (shamil-gadelshin) [inline runtime-modules/working-group/src/lib.rs]: Please, change this auto-generated comment to `Working group WeightInfo`.

File: runtime-modules/working-group/src/lib.rs [hunk 4]
```diff
@@ -469,7 +563,23 @@ decl_module! {
...
+        ///
+        /// ## Weight
+        /// `O (P)` where:
+        /// - `P` is the length of `penalty.slashing_text`
+        /// - DB:
+        ///    - O(1) doesn't depend on the state or parameters
+        /// # </weight>
+        #[weight = WeightInfoWorkingGroup::<T, I>::terminate_role_lead(
+                    penalty.as_ref().map(|penalty| penalty.slashing_text.len().saturated_into())
+                    .unwrap_or_else(|| 0)
```
Comment (shamil-gadelshin) [inline runtime-modules/working-group/src/lib.rs]: Please, change to unwrap_or_default()

File: runtime-modules/working-group/src/lib.rs [hunk 5]
```diff
@@ -469,7 +563,23 @@ decl_module! {
...
+                )
+                .max(WeightInfoWorkingGroup::<T, I>::terminate_role_worker(
+                        penalty.as_ref().map(|penalty| penalty.slashing_text.len().saturated_into())
+                            .unwrap_or_else(|| 0)
```
Comment (shamil-gadelshin) [inline runtime-modules/working-group/src/lib.rs]: Please, change to unwrap_or_default()

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]

**Review by shamil-gadelshin — APPROVED** [review]
LGTM

**Pull request merged**